src/client/Microsoft.Identity.Client/Platforms/netdesktop/NetDesktopCryptographyManager.cs

```diff
@@ -13,11 +13,18 @@
 using Microsoft.Identity.Client.Utils;
 using Microsoft.Win32.SafeHandles;
 using Microsoft.Identity.Client.Platforms.net45.Native;
+using System.Collections.Concurrent;
+using System.Threading;
 
 namespace Microsoft.Identity.Client.Platforms.net45
 {
     internal class NetDesktopCryptographyManager : ICryptographyManager
     {
+        private static ConcurrentDictionary<string, RSA> CertificateToRsaMap = new ConcurrentDictionary<string, RSA>();
```

Comment:
How large are these RSA  objects? Do we really need this limit?

Reply:
When we designed other caches, we assumed that the count would be small. 
I turned out, as usual, some scenarios we hadn't thought of, filled the cache up with millions of object, services had to be restarted periodically.

This is sort of a fail safe.